doc/adr/0008-support-inline-datums-in-constraint-library.rst

```diff
@@ -0,0 +1,52 @@
+.. _support_inline_datums_in_constraint_library:
+
+ADR 8: Support inline datums in constraint library
+==================================================
+
+Date: 2022-08-14
+
+Author(s)
+---------
+
+koslambrou <konstantinos.lambrou@iohk.io>
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+In Babbage era (available after the Vasil HF), the Cardano blockchain will support `inline` datums by changing the ``TxOut`` data type.
+
+In Alonzo era, a ``TxOut`` was able to store arbitrary data called the datum.
+However, only the hash of the datum was stored, not the actual datum.
+
+With inline datums available in Babbage era, transaction outputs can either contain the hash of the datum or the actual datum.
+Thus, we need to adapt our transaction constraint data type (``TxConstraints``) to support this new feature.
+
+Decision
+--------
+
+* We will replace the ``Datum`` parameter in ``TxConstraints``'s data constructor ``MustPayToPubKeyAddress`` with ``Plutus.V2.Ledger.Api.OutputDatum``.
+  In the offchain implementation of the constraint, we will use this new data constructor parameter to support either adding the datum in the datum witness set (by using the datum lookups to resolve the hash) or inline it in the transaction output.
+  In the PlutusV1 on-chain implementation of the constraint, we will return ``False`` if the datum value matches ``OutputDatum Datum`` because the ledgers forbids using Babbage era features with PlutusV1.
```

Comment:
Suggest "ledgers forbid" if ledgers is actually intended to be understood as plural. Or in the case of singular "ledger" then "ledger forbids".

Reply:
```suggestion
  In the PlutusV1 on-chain implementation of the constraint, we will return ``False`` if the datum value matches ``OutputDatum Datum`` because the ledger forbids using Babbage era features with PlutusV1.
```